src/components/lib/cos_kernel_api.c

```diff
@@ -149,6 +158,62 @@ __untyped_bump_alloc(struct cos_compinfo *ci)
 	return __mem_bump_alloc(ci, 1, 0);
 }
 
+static int
+__cos_mem_alias_at(struct cos_compinfo *dstci, vaddr_t dst, struct cos_compinfo *srcci, vaddr_t src, int superpage)
+{
+	int order;
+	assert(srcci && dstci);
+
+	order = (superpage) ? SUPER_PAGE_ORDER : PAGE_ORDER;
+
+	if (call_cap_op(srcci->pgtbl_cap, CAPTBL_OP_CPY, src, dstci->pgtbl_cap, dst, order)) BUG();
+
+	return 0;
+}
+
+static vaddr_t
+__superpage_mem_bump_alloc(struct cos_compinfo *ci, vaddr_t addr, int order)
+{
+	vaddr_t              inc, ret = 0;
+	vaddr_t              *ptr, frontier;
+
+	ptr      = &ci->mi.super_ptr;
+	frontier = ci->mi.super_frontier;
+
+	/* Note: this creates a hole! */
+	if (order == SUPER_PAGE_ORDER) {
+	    inc = round_up_to_pgd_page(*ptr + 1) - *ptr; /* RSK -- get the diff to add */
+	} else {
+		inc = 1 << order;
+	}
+	ret = ps_faa(ptr, inc);
+
+	/* RSK -- sanity check */
+    assert(*ptr % (1 << order) == 0);
+	if (ret >= frontier) return 0;
+
+	if (call_cap_op(BOOT_CAPTBL_SELF_UNTYPED_PT, CAPTBL_OP_MEMACTIVATE, *ptr, BOOT_CAPTBL_SELF_PT, addr, order)) return 0;
+
+	return ret;
+}
+
+/* NOTE: Not advisable to call this in non-booter components such as FWPs. If you would like to be able to use this
+ * functionality in something other than a booter (DPDK, FWP Manager), please talk to @rskennedy! */
+static vaddr_t
```

Comment:
The comment here worries me.  Can we fix this so that it works generically?  If not, what should we replace it with?